addons/mail/models/mail_thread.py

```diff
@@ -4517,6 +4517,8 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
                 if body or not message._filter_empty()
                 else ""
             )
+        if not (body and attachment_ids and partner_ids):
+            msg_values["parent_id"] = False
```

Comment:
This change (as well as my other comment) might only be acceptable in channel (and not generic thread), as `parent_id` is used differently in chatter/other threads.